yeonjae/programmers-1/체육복.js

```diff
@@ -0,0 +1,36 @@
+function solution(n, lost, reserve) {
+  //체육복을 입을 수 있는 학생 수
+  let answer = n;
+
+  //전체학생이 1개씩 가지고있다고 가정
+  let totalStudent = new Array(n).fill(1);
+
+  //totalStudent배열에서 잃어버린학생은 -1, 여유분이 있는학생은 +1
+  for (let i = 0; i < totalStudent.length; i++) {
+    if (lost.includes(i + 1)) {
+      totalStudent[i] -= 1;
+    }
+    if (reserve.includes(i + 1)) {
+      totalStudent[i] += 1;
+    }
+  }
+
+  //여유분이 있으면서 옆학생한테 빌려줄수있는경우
+  for (let i in totalStudent) {
+    if (totalStudent[i] == 2 && totalStudent[i - 1] == 0) {
+      totalStudent[i] -= 1;
+      totalStudent[i - 1] += 1;
+    } else if (totalStudent[i] == 0 && totalStudent[i - 1] == 2) {
+      totalStudent[i] += 1;
+      totalStudent[i - 1] -= 1;
+    }
+  }
+
+  //전체에서 체육복이 없는 친구 제거
+  for (let student of totalStudent) {
+    if (student === 0) {
+      answer--;
+    }
+  }
+  return answer;
+}
```

Comment:
```suggestion
function solution(n, lost, reserve) {
  //전체학생이 1개씩 가지고있다고 가정
  const totalStudent = new Array(n).fill(1);
  //체육복을 입을 수 있는 학생 수
  let answer = n;

  // totalStudent배열에서 잃어버린학생은 -1, 여유분이 있는학생은 +1
  totalStudent.forEach((_, i) => {
    if (lost.includes(i + 1)) {
      totalStudent[i] -= 1;
    }
    if (reserve.includes(i + 1)) {
      totalStudent[i] += 1;
    }
  });

  //여유분이 있으면서 옆학생한테 빌려줄수있는경우
  totalStudent.forEach((_, i) => {
    const curItem = totalStudent[i];
    const prevItem = totalStudent[i - 1];

    if (curItem === 2 && prevItem === 0) {
      totalStudent[i] -= 1;
      totalStudent[i - 1] += 1;
    } else if (curItem === 0 && prevItem === 2) {
      totalStudent[i] += 1;
      totalStudent[i - 1] -= 1;
    }
  });

  return totalStudent.filter((el) => el).length;
}
```
- 풀이 잘 봤습니다 totalStudent는 배열이고 재할당을 하지 않기 때문에 const로 수정해줬습니다. 또한, 위에 풀이는 배열 메서드들을 최대한 활용하고, 좀 변수로 빼도 괜찮아 보이는 부분을 변수로 빼서 코드길이를 최적화 해봤습니다 🙇‍♂️